src/main/java/com/google/sps/scheduler/Scheduler.java

```diff
@@ -37,20 +38,34 @@
  * A new instance should be created for each scheduling.
  */
 public class Scheduler {
-  public static final int START_HOUR = 9;
-  public static final int START_MINUTE = 0;
-  public static final int END_HOUR = 18;
-  public static final int END_MINUTE = 0;
+  public static final int DEFAULT_START_HOUR = 9;
+  public static final int DEFAULT_START_MINUTE = 0;
+  public static final int DEFAULT_END_HOUR = 18;
+  public static final int DEFAULT_END_MINUTE = 0;
   public static final long DEFAULT_DURATION_IN_MILLISECONDS = TimeUnit.MINUTES.toMillis(30);
   private List<ExtendedTask> tasks;
   private String timeZone;
+  private final int startHour;
+  private final int startMin;
+  private final int endHour;
+  private final int endMin;
   private TreeMultimap<Long, ExtendedTask> longestFirstOrderedTasks;
   private Set<Event> orderedCalendarEvents;
   private List<ExtendedTask> scheduledTasks;
 
+
   public Scheduler(Collection<Event> calendarEvents, List<ExtendedTask> tasks, String timeZone) {
+    this(calendarEvents, tasks, timeZone, new WorkingHours(DEFAULT_START_HOUR, DEFAULT_START_MINUTE, DEFAULT_END_HOUR, DEFAULT_END_MINUTE));
+  }
+
+  // Todo: consider using builder pattern
+  public Scheduler(Collection<Event> calendarEvents, List<ExtendedTask> tasks, String timeZone, WorkingHours workingHours) {
     this.tasks = tasks;
     this.timeZone = timeZone;
+    this.startHour = workingHours.getStartHour();
+    this.startMin = workingHours.getStartMin();
+    this.endHour = workingHours.getEndHour();
+    this.endMin = workingHours.getEndMin();
```

Comment:
why don't you just store a WorkingHours instance on the instance? Like this you basically put everything into an object just to extract it out of it again here.

Reply:
I think it was done to make it easier to split work with Stefano. Also maybe it makes a bit easier to handle the Scheduler constructor, because this way it has 3 less parameters. Would your suggestion be then to just create just two LocalTime objects?

Reply:
yeah exactly I'd just store two LocalTime objects in this and store the whole thing in the Scheduler when constructing

Reply:
Due to time constraints I will leave it now as is.